mypy/fastparse.py

```diff
@@ -504,11 +526,30 @@ def translate_stmt_list(
             mark_block_unreachable(block)
             return [block]
 
+        stack = self.class_and_function_stack
+        if self.strip_function_bodies and len(stack) == 1 and stack[0] == "F":
+            return []
+
         res: list[Statement] = []
         for stmt in stmts:
             node = self.visit(stmt)
             res.append(node)
 
+        if (
+            self.strip_function_bodies
+            and can_strip
+            and stack[-2:] == ["C", "F"]
+            and not is_possible_trivial_body(res)
+        ):
+            # We only strip method bodies if they don't assign to an attribute, as
+            # this may define an attribute which has an externally visible effect.
+            visitor = FindAttributeAssign()
```

Comment:
Would it be possible to strip statements after last assignment to attribute? This could make a bit more perf gain.

Reply:
This may be possible! I'd rather not do it in this PR to keep it simple (and the impact is probably pretty minor), but it's seems like a promising follow-up improvement to investigate.